src/main.rs

```diff
@@ -63,72 +62,88 @@ fn main() {
                         let ts:String = format!("{}", t.trim() );
                         
                         match Input::from(ts) {
-                        Input::WebCam{ nth } => { 
-                            try!(api.send_photo(
-                                    m.chat.id(),
-                                    path_to_picture.clone(), // Path
-                                    Some(caption_to_picture.clone()), // caption
-                                    None, // reply_to_message_id
-                                    None  // reply_markup
-                            ));
+                        Input::WebCam{ nth } => {
+                            let cams = sac.get_webcams().iter();
+                            let w = match nth {
+                                Some (nth) => {
+                                    if nth >= cams.len() {
+                                        try!(send_message(&api, m.chat.id(),
+                                            format!("You requested the webcam #{}, but there are just {}", nth, cams.len())
+                                        ));
+                                        return Ok(ListeningAction::Continue);
+                                    } else {
+                                        let mut n = 0;
+                                        cams.filter(|_| {
+                                            n == nth
+                                        })
+                                    }
+                                },
+                                None => {
+                                    cams.filter(|_| { true })
```

Comment:
I am having trouble figuring out how to compile this. :stuck_out_tongue_closed_eyes: